Tests/FluentAssertions.Specs/Types/TypeAssertionSpecs.HaveAccessModifier.cs

```diff
@@ -58,6 +58,36 @@ public void When_asserting_an_internal_type_is_internal_it_succeeds()
             act.Should().NotThrow();
         }
 
+        [Fact]
+        public void When_asserting_an_internal_member_is_internal_it_succeeds()
```

Comment:
We should also test enums and structs.
You could modify
`When_asserting_an_internal_type_is_internal_it_succeeds` and `When_asserting_an_internal_type_is_protected_internal_it_throws`
from a `[Fact]` to a `[Theory]` covering `class`, `interface`, `enum`, `struct`.